akka-persistence-query/src/test/scala/akka/persistence/query/DummyReadJournal.scala

```diff
@@ -72,3 +72,23 @@ class DummyReadJournalProvider4(@unused sys: ExtendedActorSystem, @unused conf:
 class DummyReadJournalProvider5(@unused sys: ExtendedActorSystem) extends DummyReadJournalProvider
 
 class CustomDummyReadJournalProvider5(@unused sys: ExtendedActorSystem) extends DummyReadJournalProvider("custom")
+
+class DummyReadJournalProvider7(@unused sys: ExtendedActorSystem, conf: Config, @unused confPath: String)
+    extends DummyReadJournalProvider {
+
+  // This key is can't be found in the conf, and is required by our plugin
+  val customValue: String = conf.getString("custom-key")
+
+}
+
+class DummyReadJournalProvider8(
+    @unused sys: ExtendedActorSystem,
+    @unused conf: Config,
+    @unused confPath: String,
+    mergedConf: Config)
+    extends DummyReadJournalProvider {
+
+  // This key is only found in the mergedConf, and is required by our plugin
+  val customValue: String = mergedConf.getString("custom-key")
```

Comment:
please put a comment here about the custom-key being a "root" key from the custom config, that it is not inside of the plugin config

There's some potentially very confusing consequences of this, depending on what settings the plugin reads, but since we already halfways added this feature I think we'll just have to accept that.

Reply:
Done.

Reply:
Wouldn't it be possible to solve this with:
```
ConfigFactory.parseString(s"""
       foo {
          class = "clzz"
          custom-key = $${root-custom-key}
        }
        root-custom-key = "custom-value"
        """).resolve()
```

Then the plugin still has access to `custom-key` since it's inside the pluginConfig section?

Reply:
@rdubois ^ ?

Reply:
Sorry, I didn't see your feedback. I'll take a look.

Reply:
It's interesting...

It will not prevent us from making a change on the plugins side, but it would allow to make make the change in `akka`.

Example on `akka-persistence-jdbc`:

FROM:
```
private val writePluginId = pluginConfig.getString("write-plugin")
private val eventAdapters = Persistence(system).adaptersFor(writePluginId)
```
TO (or similar):
```
private val writePluginId = pluginConfig.getString("write-plugin")
private val eventAdapters = Persistence(system).adaptersFor(writePluginId, config)
```

and on the client side, the referenced `write-plugin` should be injected into the `read-journal` plugin configuration.

@patriknw, do you confirm this was what you had in mind ? And do you also confirm that this alternative solution will replace the actual PR ?

Reply:
@patriknw can you please take a look at my previous message ? Thanks.

Reply:
Patrik is on vacation, so it may be a while before he can respond, just FYI

Reply:
Hello @patriknw, @johanandren, when you have time, can you please review the question above ? Thanks.

Reply:
Sorry for delays. Looking at this again. I think the easiest way forward would be if my suggestion here would work for what you need. Advantages:
* Then we wouldn't have to add yet another constructor parameter variant.
* We don't have to backport and release this for Akka 2.5.x. Currently we haven't scheduled a release for 2.5.x.
* Less risk that we introduce another feature for this dynamic plugin config that must be revised anyway later. There is already something in the Cassandra plugin that isn't working https://github.com/akka/akka-persistence-cassandra/issues/684

Reply:
Hi @patriknw,

I'll try to put that in placce then, but would you mind to confirm my assumptions above, regarding the changes to be done on plugins side ? https://github.com/akka/akka/pull/28935#discussion_r417089539

Reply:
I don't know enough about the config in akka-persistence-jdbc to directly say if that is right or wrong. Open a PR there and we'll review. One challenge with all this config juggling is that some are from the root level and some are only the plugin section. Previous behavior in akka-persistence-jdbc when using it without the `readJournalPluginConfig` parameter must still work.